src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/npathcomplexity/InputNPathComplexityPatternMatchingForSwitch.java

```diff
@@ -0,0 +1,105 @@
+/*
+NPathComplexity
+max = 1
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.metrics.npathcomplexity;
+
+public class InputNPathComplexityPatternMatchingForSwitch {
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testPatternWithRule(Object o) {
+        switch (o) {
+            case Integer _ -> {}
+            case String _ -> {}
+            default -> {}
+        }
+    }
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testPatternWithStatement(Object o) {
+        switch (o) {
+            case Integer _: {}
+            case String _: {}
+            default : {}
+        }
+    }
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testRecordPatternWithRule(Object o) {
+        switch (o) {
+            case A(int x) -> {}
+            case B(String s) -> {}
+            default -> {}
+        }
+    }
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testRecordPatternWithStatement(Object o) {
+        switch (o) {
+            case A(int x) : {} break;
+            case B(String s) : {}
+            default : {}
+        }
+    }
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testGuardsInRule(Object o) {
+        switch (o) {
+            case Integer i when i > 0 -> {}
+            case String s when s.length() > 0 -> {}
+            default -> {}
+        }
+    }
+
+    // violation below, 'NPath Complexity is 3 (max allowed is 1)'
+    void testGuardsInStatement(Object o) {
```

Comment:
I think there may be more to this. I rewrote the code as it I thought it would appear without `when` and got a different number.

````
$ cat TestClass.java
public class TestClass {
    void testGuardsInStatement(Object o) {
        switch (o) {
            case Integer i when i > 0 : {} break;
            case String s when s.length() > 0 : {}
            default : {}
        }
    }

    void testGuardsInStatement(Object o) {
        switch (o) {
            case Integer i : { if (i > 0) { } } break;
            case String s : { if (s.length() > 0) { } }
            default : {}
        }
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>

    <module name="TreeWalker">
    <module name="NPathComplexity">
      <property name="max" value="0"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-10.17.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[WARN] TestClass.java:2:5: NPath Complexity is 3 (max allowed is 0). [NPathComplexity]
[WARN] TestClass.java:10:5: NPath Complexity is 5 (max allowed is 0). [NPathComplexity]
Audit done.
````

Reply:
Installed a nightly (checkstyle-nightly-2024-07-15-all.jar) just to make sure and there was no changes to counts.

https://checkstyle.org/checks/metrics/npathcomplexity.html#Description
According to documentation, ifs and switches are counted.
> `if ([expr]) { [if-range] } 	NP(if-range) + 1 + NP(expr)`
> `switch ([expr]) { case : [case-range] default: [default-range] } 	S(i=1:i=n)NP(case-range[i]) + NP(default-range) + NP(expr)`

I think we should update documentation somehow to showcase when.

Reply:
https://github.com/checkstyle/checkstyle/issues/15445